server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -342,6 +342,10 @@ public void resetBChangedForRootComponentUuid(DbSession session, String projectU
     mapper(session).resetBChangedForRootComponentUuid(projectUuid);
   }
 
+  public void setPrivateForRootComponentUuid(DbSession session, String projectUuid, boolean isPrivate) {
```

Comment:
provided uuid must be the one of a root component, otherwise calling this method is pointless, it will have no effect